e2e/cypress/integration/channel_sidebar/DM_category_spec.js

```diff
@@ -0,0 +1,107 @@
+
+const SpaceKeyCode = 32;
+const DownArrowKeyCode = 40;
+
+describe('MM-T3156 DM category', () => {
+    const sysadmin = getAdminAccount();
+    let testUser;
+    const usernames = [];
+    before(() => {
+        // # Enable channel sidebar organization
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ExperimentalChannelSidebarOrganization: 'default_on',
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup({loginAfter: true}).then(({team, user}) => {
+            testUser = user;
+            cy.visit(`/${team.name}/channels/town-square`);
+
+            // # upgrade user to sys admin role
+            cy.externalRequest({user: sysadmin, method: 'put', path: `users/${user.id}/roles`, data: {roles: 'system_user system_admin'}});
+        });
+
+        // # Close "What's new" modal
+        cy.uiCloseWhatsNewModal();
+    });
+
+    it('MM-T3156_1 Should open DM modal on click of + in category header', () => {
+        cy.get('button[aria-label="DIRECT MESSAGES"]').parents('.SidebarChannelGroup').within(() => {
+            cy.get('.SidebarChannelGroupHeader_addButton').click();
+        });
+        cy.get('#moreDmModal').should('be.visible');
+        cy.get('#moreDmModal .close').click();
+    });
+
+    it('MM-T3156_2 should order DMs based on recent interactions', () => {
+        const usersPrefixes = ['a', 'c', 'd', 'j', 'p', 'u', 'x', 'z'];
+        usersPrefixes.forEach((prefix) => {
+            // # Create users with prefixes in alphabatical order
+            cy.apiCreateUser({prefix}).then(({user: newUser}) => {
+                cy.apiCreateDirectChannel([testUser.id, newUser.id]).then((res) => {
+                    const channel = res.body;
+
+                    // # Post messaege in The DM channel
+                    cy.postMessageAs({sender: newUser, message: 'test', channelId: channel.id});
+
+                    // add usernames in array for reference
+                    usernames.push(newUser.username);
+                });
+            });
+        });
+
+        // get DM category group
+        cy.get('button[aria-label="DIRECT MESSAGES"]').parents('.SidebarChannelGroup').within(() => {
+            const usernamesReversed = [...usernames].reverse();
+
+            cy.get('.NavGroupContent').children().each(($el, index) => {
+                // * Verify that the usernames are in reverse order i.e ordered by recent activity
+                cy.wrap($el).find('.SidebarChannelLinkLabel').should('contain', usernamesReversed[index]);
+            });
+        });
+    });
+
+    it('MM-T3156_3 should order DMs alphabatically ', () => {
+        // get DM category group
+        cy.get('button[aria-label="DIRECT MESSAGES"]').parents('.SidebarChannelGroup').within(() => {
+            // # Change sorting to be alphabetical
+            cy.get('.SidebarChannelGroupHeader_sortButton').invoke('show').click();
```

Comment:
This test is intermittently failing on me but when I added wait time, it consistently passing.
```suggestion
            cy.get('.SidebarChannelGroupHeader_sortButton').invoke('show').wait(TIMEOUTS.HALF_SEC);
```

Reply:
wait and click()?

Reply:
Ooopss, sorry. Click then wait. I updated the suggestion.